go/adbc/driver/databricks/schema.go

```diff
@@ -128,182 +194,287 @@ func ResultSchemaToArrowSchema(dbxSchema *sql.ResultSchema) (*arrow.Schema, erro
 	return arrow.NewSchema(fields, &metadata), nil
 }
 
-func getArrowTypeFromColumnInfo(col sql.ColumnInfo) (arrow.DataType, error) {
-	if arrowType, ok := basicTypeToArrowTypeMap[col.TypeName]; ok {
-		return arrowType, nil
+// extracts the content between the outermost <...> after the given prefix.
+func extractOutermostBracketContent(s, prefix string) (string, bool) {
+	if !strings.HasPrefix(s, prefix+"<") || !strings.HasSuffix(s, ">") {
+		return "", false
 	}
-	switch col.TypeName {
-	case sql.ColumnInfoTypeNameDecimal:
-		precision, scale := col.TypePrecision, col.TypeScale
-		if precision == 0 {
-			// Mostly used by the recursive case (array, map, etc)
-			// "DECIMAL(p,s) and DECIMAL(p) with scale default at 0"
-			if matches := regexp.MustCompile(`DECIMAL\((\d+)(?:,?(\d+))?\)`).FindStringSubmatch(col.TypeText); matches != nil {
-				var err error
-				var precision64 int64
-				precision64, err = strconv.ParseInt(matches[1], 10, 8)
-				if err != nil {
-					return nil, fmt.Errorf("invalid decimal precision: %v", err)
-				}
-				precision = int(precision64)
-
-				// expect 3 matches total: DECIMAL, p, s
-				if len(matches) == 3 {
-					var scale64 int64
-					scale64, err = strconv.ParseInt(matches[2], 10, 8)
-					if err != nil {
-						return nil, fmt.Errorf("invalid decimal scale: %v", err)
-					}
-					scale = int(scale64)
-				}
+	start := len(prefix) + 1
+	depth := 0
+	for i := start; i < len(s); i++ {
+		if s[i] == '<' {
+			depth++
+		} else if s[i] == '>' {
+			if depth == 0 {
+				// Found the matching closing bracket
+				return s[start:i], true
 			}
+			depth--
 		}
-		if precision <= 38 {
-			return &arrow.Decimal128Type{Precision: int32(precision), Scale: int32(scale)}, nil
-		}
-		return &arrow.Decimal256Type{Precision: int32(precision), Scale: int32(scale)}, nil
-	// todo(jasonlin45): Add support for TIMESTAMP_NTZ
-	case sql.ColumnInfoTypeNameTimestamp:
+	}
+	// If we get here, brackets were unbalanced
+	return "", false
+}
+
+func getArrowTypeFromStringType(col string) (arrow.DataType, error) {
+	// simple types - base case
+	if arrowType, ok := stringTypeToArrowTypeMap[col]; ok {
+		return arrowType, nil
+	}
+	if col == "TIMESTAMP" || col == "TIMESTAMP_NTZ" {
```

Comment:
why is timestamp logic here? I know it's not always worth jumping around but `if col == "TIMESTAMP" || col == "TIMESTAMP_NTZ" {` feels more general even if it's only ever used once. I don't see the logic in only having it here except to be DRY